pkg/kubelet/kuberuntime/labels.go

```diff
@@ -193,6 +200,11 @@ func getContainerInfoFromAnnotations(annotations map[string]string) *annotatedCo
 	if containerInfo.Hash, err = getUint64ValueFromLabel(annotations, containerHashLabel); err != nil {
 		klog.ErrorS(err, "Unable to get label value from annotations", "label", containerHashLabel, "annotations", annotations)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		if containerInfo.HashImmutableFields, err = getUint64ValueFromLabel(annotations, containerHashImmutableFieldsLabel); err != nil {
+			klog.ErrorS(err, "Unable to get label value from annotations", "label", containerHashImmutableFieldsLabel, "annotations", annotations)
```

Comment:
This seems good to me, I think this is cleaner than the previous implementation.